lib/utils/initializeAxios.js

```diff
@@ -0,0 +1,33 @@
+const axios = require('axios');
+const qs = require('querystring');
+
+const initializeAxios = (config, production = false) => {
+    const { username, apiKey, format } = config;
+    const baseURL = production
+        ? 'https://api.africastalking.com/version1/'
+        : 'https://api.sandbox.africastalking.com/version1';
+
+    const createAxiosInstance = (contentType = 'application/x-www-form-urlencoded') => {
+        return axios.create({
+            baseURL,
+            headers: {
+                apiKey,
+                'Content-Type': contentType,
+                Accept: format,
+                data: null,
+            },
+        });
+    };
+
+    return {
```

Comment:
I think these functions should live inside their respective modules. This looks fine now but will be quite crowded when you move all methods.

Reply:
Hi Calvin,
I see your point and couldn't help but agree.

However, in order to achieve modularity/separation of concerns, wouldn't it be better to keep the functions for API requests and the respective modules _separate_?

Inherently, the functions returned by the `initializeAxios` file have several dependencies which are not needed by the respective modules. 

For example:
- some requests need the `querystring` module which is used to do url-encoding before sending an API request
- some requests need the api `username` attribute in the POST data field

Moving these functions to the respective modules will mean redundant repetitive imports for each of the respective modules, which is what I was trying to avoid.

Reply:
I see what you mean. The real issue is this module getting bloated, maybe you can group related functions in their own modules, this module could then look like:

```javascript
const initializeAxios = (..., product) => {

    ...

   const createAxiosInstance = () => { ... };

    /*
    * productFunctions can be a module that re-exports
    * an object with all product specific functions
    *
    * import airtime from ...
    * import application from ...
    * 
    * export default { airtime, application };
    */
   return productFunctions[product](createAxiosInstance);
}
```

Reply:
Hi @calvinkarundu 

This is now resolved.